### PR TITLE
Bump all things

### PR DESCRIPTION
**Pull request by Goooler**

### CODE REVIEW

**Review by simon300000 — APPROVED** [review]
lgtm

**Comment by Goooler** [comment]
等下回去我把 ci 修好

**Review by simon300000 — CHANGES_REQUESTED** [review]
应该把每个改动分一个单独的pr

**Comment by Goooler** [comment]
@ilharp CC

File: .editorconfig [hunk 1]
```diff
@@ -5,11 +5,7 @@ insert_final_newline = true
 trim_trailing_whitespace = true
 indent_style = space
 indent_size = 2
-end_of_line = lf
 charset = utf-8
 
-[gradlew.bat]
-end_of_line = crlf
-
```
Comment (ilharp) [inline .editorconfig:13]: `.editorconfig` should keep the same behavior as `.gitattributes`.

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -7,19 +7,23 @@ on:
...
+    - name: Validate wrapper
+      uses: gradle/wrapper-validation-action@v1
+
+    - name: Set-up JDK 17
+      uses: actions/setup-java@v3
       with:
-        distribution: 'temurin'
-        java-version: '11'
+        distribution: 'zulu'
+        java-version: '17'
```
Comment (ilharp) [inline .github/workflows/build.yml:26]: According to [WhichJDK](https://whichjdk.com), [Adoptium Temurin](https://adoptium.net) should be the best choice. Why zulu?

File: gradle.properties [hunk 3]
```diff
@@ -1,2 +1,6 @@
 kotlin.code.style=official
 kotlin.js.generate.executable.default=false
+
+org.gradle.caching=true
+org.gradle.jvmargs=-Xmx4g -Dfile.encoding=UTF-8
+org.gradle.parallel=true
```
Comment (ilharp) [inline gradle.properties:6]: Please give explanations of these parameters separately. The target machine to run this build on is unknown, so we should not specify memory parameters.

File: settings.gradle.kts [hunk 4]
```diff
@@ -1 +1,28 @@
...
+    repositories {
+        gradlePluginPortal()
+        mavenCentral()
+    }
+}
+
+dependencyResolutionManagement {
+    repositories {
+        google()
+        maven("https://maven.aliyun.com/repository/public")
```
Comment (ilharp) [inline settings.gradle.kts:11]: Do not use mirror repository in public source code.

File: settings.gradle.kts [hunk 5]
```diff
@@ -1 +1,28 @@
...
+}
+
+gradleEnterprise {
+    buildScan {
+        termsOfServiceUrl = "https://gradle.com/terms-of-service"
+        termsOfServiceAgree = "yes"
+        publishAlwaysIf(System.getenv("GITHUB_ACTIONS") == "true")
+        publishOnFailure()
+    }
+}
```
Comment (ilharp) [inline settings.gradle.kts:26]: I didn't see the benefit of enabling `gradle-enterprise` on such a small project.

**Review by ilharp — CHANGES_REQUESTED** [review]
The same view as @simon300000. Divide your PR according to changes and resubmit. One single change in one PR.

**Pull request closed without merging**